### ISSUE TITLE
Feature request: Unset SPC means `y` when prompted for `y` or `n`

### ISSUE THREAD

**Issue by dalanicolai** [syl20bnr/spacemacs#14382]
Personally I find the behavior that pressing `SPC` means `y` when prompted for `y` or `n` quite annoying. The behavior makes sense for vanilla Emacs but is not "safe" in Spacemacs. This behavior can simply be changed by placing `(define-key query-replace-map (kbd "SPC") nil)` in the `dotspacemacs/user-config`. I have unsuccessfully tried to implement it as a `dotspacemacs-` variable. So I decided to just open an issue for it.

Test:
`M-: (y-or-n-p)` and press `SPC`

**Comment by smile13241324** [comment]
I have discussed that with the others especially I wanted to know whether this behaviour was intentional or not. Turns out it isn't. I will add a new var to the dotfile to control this behaviour. Something among the line of `use-SPC-as-y` default `nil`. I think it makes sense to assume that most people will use `SPC` as leader key hence using it also to confirm possibly important actions is just dangerous and should be changed.
